incubator/binding-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/asyncapi/internal/config/AsyncapiBindingConfig.java

```diff
@@ -235,7 +239,9 @@ private void attachServerClientBinding(
         for (AsyncapiSchemaConfig config : configs)
         {
             Asyncapi asyncapi = config.asyncapi;
-            final NamespaceConfig composite = binding.attach.apply(namespaceGenerator.generate(binding, asyncapi));
+            final NamespaceConfig composite = namespaceGenerator.generate(binding, asyncapi);
+            composite.readURL = binding.readURL;
```

Comment:
I wonder if we still need this since we don't use anymore readURL

Reply:
Yep, per our discussion there is still an `assert` for non-null `readURL` on namespace in engine.
Looking forward to perhaps being able to remove this later when we only use catalog and no longer need direct local file access from bindings, but still needed for now.